Tools/TestWebKitAPI/Tests/WebCore/cg/GraphicsContextTestsCG.cpp

```diff
@@ -74,6 +75,29 @@ TEST(GraphicsContextTests, DrawNativeImageDoesNotLeakCompositeOperator)
     EXPECT_EQ(ctx.blendMode(), BlendMode::Normal);
 }
 
+TEST(GraphicsContextTests, CGBitmapRenderingModeIsCorrect)
+{
+    auto srgb = DestinationColorSpace::SRGB();
+    auto cgContext = adoptCF(CGBitmapContextCreate(nullptr, 3, 3, 8, 4 * 3, srgb.platformColorSpace(), kCGImageAlphaPremultipliedLast));
+    ASSERT_NE(cgContext, nullptr);
+    GraphicsContextCG context(cgContext.get());
+    EXPECT_EQ(context.renderingMode(), RenderingMode::Unaccelerated);
+}
+
+TEST(GraphicsContextTests, IOSurfaceRenderingModeIsCorrect)
+{
+    auto srgb = DestinationColorSpace::SRGB();
+    IntSize size { 3, 3 };
+    auto surface = IOSurface::create(nullptr, size, srgb);
+    ASSERT_NE(surface, nullptr);
+    auto bitsPerPixel = 32;
+    auto bitsPerComponent = 8;
+    auto bitmapInfo = static_cast<CGBitmapInfo>(kCGImageAlphaPremultipliedFirst | kCGBitmapByteOrder32Host);
+    auto cgContext = adoptCF(CGIOSurfaceContextCreate(surface->surface(), size.width(), size.height(), bitsPerComponent, bitsPerPixel, srgb.platformColorSpace(), bitmapInfo));
+    GraphicsContextCG context(cgContext.get());
+    EXPECT_EQ(context.renderingMode(), RenderingMode::Accelerated);
+}
```

Comment:
It is strange that this test creates a context with size = { 3, 3 } and `context.renderingMode()` still returns `Accelerated`. Is there a way to get and test the case when this context is unaccelerated?

Reply:
Yes, that's the case above.
If the context is IOSurface context, it's accelerated.
If the context is CGBitmap context, it's unaccelerated.
Size has nothing to do with it.

Reply:
Added a corresponding test for the small and large CALayer